src/java_bytecode/java_bytecode_convert_method.cpp

```diff
@@ -263,7 +263,41 @@ void java_bytecode_convert_method_lazy(
   symbol_tablet &symbol_table)
 {
   symbolt method_symbol;
-  typet member_type=java_type_from_string(m.descriptor);
+
+  typet member_type;
+  if(m.signature.has_value())
+  {
+    try
+    {
+      typet member_type_s=java_type_from_string(
+        m.signature.value(),
+        id2string(class_symbol.name));
+      INVARIANT(member_type_s.id()==ID_code, "Must be code type");
+      typet member_type_d=java_type_from_string(
+        m.descriptor,
+        id2string(class_symbol.name));
+      INVARIANT(member_type_d.id()==ID_code, "Must be code type");
+      if(to_code_type(member_type_s).parameters().size()==
+         to_code_type(member_type_d).parameters().size())
+      {
+        member_type=member_type_s;
+      }
+      else
+      {
+        // TODO: by reverting to descriptor, we are loosing generics info. In
+        // the future, we may want to extract that info from signature.
+        member_type=member_type_d;
+      }
+    }
+    catch(unsupported_java_class_signature_exceptiont &e)
```

Comment:
I suspect we should log a warning whenever this happens saying `"could not parse signature for method_identifier, falling back to descriptor: e.message"`